packages/slo-stat-bot/src/slo-logic.ts

```diff
@@ -0,0 +1,504 @@
+  login: string;
+  permissions: {
+    pull: boolean;
+    push: boolean;
+    admin: boolean;
+  };
+}
+
+//Checking if slo applies to a given issue
+getSLOStatus.appliesTo = async function appliesTo(
+  slo: SLORules,
+  issueLabels: string[] | null
+): Promise<boolean> {
+  if (issueLabels === null || issueLabels.length === 0) {
+    return false;
+  }
+
+  //Checking if all the githublabels are subset of issue labels
+  const githubLabels = slo.appliesTo.gitHubLabels;
+  const isValidGithubLabels = await getSLOStatus.isValidGithubLabels(
+    issueLabels,
+    githubLabels
+  );
+  if (!isValidGithubLabels) {
+    return false;
+  }
+
+  //Checking if all the excluded github labels are not in issue labels
+  const excludedGitHubLabels = slo.appliesTo.excludedGitHubLabels;
+  const isValidExcludeLabels = await getSLOStatus.isValidExcludedLabels(
+    issueLabels,
+    excludedGitHubLabels
+  );
+  if (!isValidExcludeLabels) {
+    return false;
+  }
+
+  //Checking if priority is present and matches in issue labels
+  const priority = String(slo.appliesTo.priority);
+  const isValidPriority = await getSLOStatus.isValidRule(
+    issueLabels,
+    priority,
+    'priority: '
+  );
+  if (!isValidPriority) {
+    return false;
+  }
+
+  //Checking if issue type is present and matches in issue labels
+  const issueType = slo.appliesTo.issueType;
+  const isValidIssueType = await getSLOStatus.isValidRule(
+    issueLabels,
+    issueType,
+    'type: '
+  );
+  if (!isValidIssueType) {
+    return false;
+  }
+
+  return true;
+};
+
+getSLOStatus.isValidGithubLabels = async function isValidGithubLabels(
+  issueLabels: string[],
+  githubLabels: string | string[] | undefined
+): Promise<boolean> {
+  if (!githubLabels) {
+    return true;
+  }
+
+  githubLabels = await getSLOStatus.convertToArray(githubLabels);
+  githubLabels.forEach((label: string) => label.toLowerCase());
+  const isSubSet = githubLabels.every((label: string) =>
+    issueLabels.includes(label)
+  );
+  if (!isSubSet) {
+    return false;
+  }
+  return true;
+};
+
+getSLOStatus.isValidExcludedLabels = async function isValidExcludedLabels(
+  issueLabels: string[],
+  excludedGitHubLabels: string | string[] | undefined
+): Promise<boolean> {
+  if (!excludedGitHubLabels) {
+    return true;
+  }
+
+  excludedGitHubLabels = await getSLOStatus.convertToArray(
+    excludedGitHubLabels
+  );
+  excludedGitHubLabels.forEach((label: string) => label.toLowerCase());
+  const isElementExist = excludedGitHubLabels.some((label: string) =>
+    issueLabels.includes(label)
+  );
+  if (isElementExist) {
+    return false;
+  }
+  return true;
+};
+
+getSLOStatus.isValidRule = async function isValidRule(
+  issueLabels: string[],
+  rule: string | undefined,
+  title: string
+) {
+  if (!rule) {
+    return true;
+  }
+
+  rule = rule.toLowerCase();
+  const includes =
+    issueLabels.includes(rule) || issueLabels.includes(title + rule);
+  if (!includes) {
```

Comment:
I think you could just return includes. Same for other similar functions in this file.

Reply:
That's true. I'll fix it!